### PR TITLE
Refactor documentation to markdown

### PR DESCRIPTION
**Pull request by bruno-f-cruz**
This PR refactors the main documentation documents of the Harp protocol to use markdown instead of `docx`.

This will also facilitate keep tracking of versions, so my suggestion would be to only store the markdown files and, when a release is made, attach the corresponding pdfs (which can be easily automatically generated) to the release files.

As a side note, I am not fully satisfied with the way GitHub renders `Mermaid` gantt plots. While it works nicely when rendering it using my VSCode plugin, github rendered does not allow the `compact` mode, leading bars to offset. We might want to converge on a general solution to generate these diagrams (e.g. bit maps).

### CODE REVIEW

File: BinaryProtocol-8bit.md [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+Contains the number of bytes that are still available and need to be read to complete the Harp message (i.e. number of bytes after the field [`Length`]).
+
+### [`Address`] (1 byte)
+
+Contains the address of the register to which the Harp Message refers to.
+
+### [`Port`] (1 byte)
+
+If the device is a Hub of Harp Devices, it indicates the origin or destination of the Harp Message. If the field is not used or it’s equal to `0xFFFFFFFF`, it points to the device itself.
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:75]: On the [Port] description shouldn't be 0xFF instead of 0xFFFFFFFF?

File: BinaryProtocol-8bit.md [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+#### Seconds (4 bytes)
+
+Contains the number of seconds (`U32`) of the Harp Timestamp clock. This field is optional. In order to indicate that this field is available, the bit [HasTimestamp] in the field [PayloadType] needs to be set.
+
+#### Nanoseconds (2 bytes)
+
+It contains the fractional part of the Harp Timestamp clock in nanoseconds (`U16` containing the number of microseconds divided by 32).
+
+This field is optional. In order to indicate that this field is available, the bit [HasTimestamp] in the field [PayloadType] needs to be set.
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:142]: On the Payload description, shouldn't be microseconds instead of nanoseconds?

File: BinaryProtocol-8bit.md
Comment (artursilva0) [inline BinaryProtocol-8bit.md:1]: Isn't the 8bit.md incomplete after the features and code examples?

File: Device.md [hunk 3]
```diff
@@ -0,0 +1,471 @@
...
+    23-16      :byte2, after byte3  , 2
+    15-8      :byte1, after byte2  , 3
+    7-0      :byte0, after byte1  , 4
+    section Id
+    SECONDS      :id1, 0, 4
+    section Default
+    0      :d1, 0, 4
+```
+
+#### **`R_TIMESTAMP_SECOND` (U16) – System timestamp (seconds)**
```
Comment (artursilva0) [inline Device.md:139]: R_TIMESTAMP_SECOND (U16) – System timestamp (seconds), shouldn't be microseconds?

File: Device.md [hunk 4]
```diff
@@ -0,0 +1,471 @@
...
+    15-8      :byte1, 0, 1
+    7-0      :byte0, after byte1  , 2
+    section Id
+    USECONDS      :id1, 0, 2
+    section Default
+    -      :d1, 0, 2
+```
+
+
+#### **`R_OPERATION_CTRL` (U16) – Operation mode configuruation**
```
Comment (artursilva0) [inline Device.md:167]: R_OPERATION_CTRL (U16) – Operation mode configuruation - should be configuration

File: Device.md [hunk 5]
```diff
@@ -0,0 +1,471 @@
...
+
+* **CLK_GEN [Bit 1]:** If set to 1, the device will generate Harp Synchronization Clock to the Clock Output connector, if available. The Clock Input will be ignored. The bit is read as 1 if the device is generating the Harp Synchronization Clock.
+
+* **REP_ABLE [Bit 3]:** This is a read-only bit signaling if the device is able (1) to repeat the Harp Synchronization Clock timestamp.
+
+* **GEN_ABLE [Bit 4]:** This is a read-only bit signaling if the device is able (1) to generate the Harp Synchronization Clock timestamp.
+
+* **CLK_UNLOCK [Bit 6]:** If set to 1, the device will unlock the timestamp register counter (register `R_TIMESTAMP_SECOND`) and it will now accept new timestamp values. The bit is read as 1 if the timestamp register is unlocked.
+
+* **CLK_UNLOCK [Bit 7]:** If set to 1, the device will lock the current timestamp register counter (register `R_TIMESTAMP_SECOND`) and it will reject any new timestamp values. The bit is read as 1 if the timestamp register is locked.
```
Comment (artursilva0) [inline Device.md:401]: CLK_UNLOCK [Bit 7]-> CLK_LOCK [Bit 7]

**Comment by artursilva0** [comment]
> This PR refactors the main documentation documents of the Harp protocol to use markdown instead of `docx`.
> 
> This will also facilitate keep tracking of versions, so my suggestion would be to only store the markdown files and, when a release is made, attach the corresponding pdfs (which can be easily automatically generated) to the release files.
> 
> As a side note, I am not fully satisfied with the way GitHub renders `Mermaid` gantt plots. While it works nicely when rendering it using my VSCode plugin, github rendered does not allow the `compact` mode, leading bars to offset. We might want to converge on a general solution to generate these diagrams (e.g. bit maps).

I also don't like the way those plots were generated. They are not very clear unfortunately. What do you mean by bit maps?

**Comment by bruno-f-cruz** [comment]
> > This PR refactors the main documentation documents of the Harp protocol to use markdown instead of `docx`.
> > This will also facilitate keep tracking of versions, so my suggestion would be to only store the markdown files and, when a release is made, attach the corresponding pdfs (which can be easily automatically generated) to the release files.
> > As a side note, I am not fully satisfied with the way GitHub renders `Mermaid` gantt plots. While it works nicely when rendering it using my VSCode plugin, github rendered does not allow the `compact` mode, leading bars to offset. We might want to converge on a general solution to generate these diagrams (e.g. bit maps).
> 
> I also don't like the way those plots were generated. They are not very clear unfortunately. What do you mean by bit maps?

bit maps = the maps that describe what each one of the bits represent.

Ye I am not super happy with them either, but the alternatives were not clearly better. 
HTML is way less readable;
Markdown tables are too limited;
SVGs look nicer but you need additional software to draw them and you need to save them as assets

Do you have a suggestion?

File: BinaryProtocol-8bit.md
Comment (bruno-f-cruz) [reply BinaryProtocol-8bit.md:1]: What is missing?

Comment (artursilva0) [reply BinaryProtocol-8bit.md:1]: The last pages of the equivalent file in .pdf with more code examples, typical usage and version control. Am I looking at the wrong .md? The one I can see has 185 lines.

**Comment by artursilva0** [comment]
Personally I prefer the .svg option since it's the only one that will be capable of generating any sort of diagrams that might necessary in the future. Let's see what the other reviewers think about that.

File: BinaryProtocol-8bit.md [hunk 6]
```diff
@@ -0,0 +1,347 @@
...
+|  Value  |  Description  |
+| :-----: |  -----------: |
+| 1       |    8 bits     |
+| 2       |   16 bits     |
+| 4       |   32 bits     |
+| 8       |   64 bits     |
+
+#### HasTimestamp (1 bit)
+
+If this bit is set the Harp Message contains a timestamp. In this case the fields [Seconds] and [Nanoseconds] must be present in the message.
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:126]: [Nanoseconds] -> [Microseconds]

File: BinaryProtocol-8bit.md [hunk 7]
```diff
@@ -0,0 +1,347 @@
...
+#### Microseconds (2 bytes)
+
+It contains the fractional part of the Harp Timestamp clock in microseconds (`U16` containing the number of microseconds divided by 32).
+
+This field is optional. In order to indicate that this field is available, the bit [HasTimestamp] in the field [PayloadType] needs to be set.
+
+> **Note**
+>
+> The full timestamp information can thus be retrieved using the formula:
+> Timestamp(s) = [Seconds] + [Nanoseconds] * 32 * 10-6
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:159]: [Nanoseconds] -> [Microseconds]

File: BinaryProtocol-8bit.md [hunk 8]
```diff
@@ -0,0 +1,347 @@
...
+      TimestampedS16 = (hasTimestamp | S16),
+      TimestampedU32 = (hasTimestamp | U32),
+      TimestampedS32 = (hasTimestamp | S32),
+      TimestampedU64 = (hasTimestamp | U64),
+      TimestampedS64 = (hasTimestamp | S64),
+      TimestampedFloat = (hasTimestamp | Float)
+  }
+```
+
+The field `PayloadType` has a flag on the 5th least significant bit that indicates if the time information is available on the Harp Message. The existence of this flag is useful to know if the fields [Seconds] and [Nanoseconds] are present on the Harp Message.
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:233]: [Nanoseconds] -> [Microseconds]

File: BinaryProtocol-8bit.md [hunk 9]
```diff
@@ -0,0 +1,347 @@
...
+
+The `Peripheral` device that runs the `Harp Protocol` receives `Write` and `Read` `Commands` from the `Host`, and, in turn, sends/responds with `Replies`.
+
+Some `Harp Message`s are shown below to demonstrate the typical usage of the protocol between a `Peripheral` and an `Host`. Note that, from the `Host` to the `Peripheral`, the time information is omitted in Harp Message, since this information is optional.
+We will use the following abbreviations:
+
+  - [CMD] is a Command (From the `Host` to the `Peripheral`);
+  - [RPL] is a Reply (From `Peripheral` to the `Host`)
+  - [EVT] is an Event. (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)s
+
```
Comment (artursilva0) [inline BinaryProtocol-8bit.md:293]: - [EVT] is an Event. (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)s
--> - [EVT] is an Event (A message sent from the `Peripheral` to the `Host` without a request from the `Host`)

**Review by artursilva0 — COMMENTED** [review]
minor typos

**Comment by artursilva0** [comment]
> Personally I prefer the .svg option since it's the only one that will be capable of generating any sort of diagrams that might necessary in the future. Let's see what the other reviewers think about that.

After our offline discussion, I've tried to implement a raw example of a CSS + HTML code inside an SVG, that can be rendered inside GitHub markdown. One of the disadvantages is that you'll still need to generate the SVGs as assets. Example here: https://github.com/artursilva0/harp-protocol/blob/docs-to-md/Device.md#r_operation_ctrl-u16--operation-mode-configuration

Nevertheless, taking into account that the way the GitHub is rendering these plots, may be changed in the Harp website (different CSS) and that the PDFs generated present a much more appealing Gantt plot, I'm ok in keeping this as it is.

**Review by artursilva0 — APPROVED** [review]

**Review by filcarv — APPROVED** [review]

**Pull request merged**